.github/workflows/main.yml

```diff
@@ -1,18 +1,24 @@
-name: Ruby
+name: Ruby Tests
 
-on: [push,pull_request]
+on: [push, pull_request]
 
 jobs:
-  build:
+  test:
+    name: Ruby ${{ matrix.ruby-version }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ['3.0.1', '3.1.1', '3.2.1', '3.3.1', '3.4.1']
+      fail-fast: false
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Ruby
+    - uses: actions/checkout@v4
+    - name: Set up Ruby ${{ matrix.ruby-version }}
       uses: ruby/setup-ruby@v1
```

Comment:
`ruby/setup-ruby@v1` is still the recommended one as it can be used with ruby `3.4.1` (latest stable version )
Ref: https://github.com/ruby/setup-ruby